runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -2312,7 +2312,12 @@ void gni_init(gni_nic_handle_t* nih, int cdi)
   GNI_CDM_MODES:
   - Check _FORK options for the data server
 \* -------------------------------------------------------------------------- */
-  GNI_CHECK(GNI_CdmCreate(cdi, ptag, cookie, modes, &cdm_handle));
+  //
+  // The instance ID must be node-unique.  The one we specify
+  // here is job-unique, even.
+  //
+  const int inst_id = (chpl_nodeID * comm_dom_cnt_max) + cdi;  //
```

Comment:
trailing comment?

Reply:
Argh. Thanks. I force-pushed a repair. I can't bear to leave that commit as it stood.